app/views/static/generic_object/main_custom_button_form.html.haml

```diff
@@ -20,6 +20,39 @@
           = "<#{_('Choose')}>"
       %span.help-block{"ng-show" => "angularForm.button_type.$error.required"}
         = _("Required")
+  .form-group{"ng-if"    => "vm.customButtonModel.button_type == 'ansible_playbook'",
+              "ng-class" => "{'has-error': angularForm.playbook_catalog_item.$invalid}"}
+    %label.control-label.col-md-2
+      = _("Playbook Catalog Item")
+    .col-md-8
+      %select{"name"                        => "playbook_catalog_item",
+              "ng-model"                    => "vm.customButtonModel.uri_attributes.service_template",
+              "ng-options"                  => "item.id as item.name for item in vm.templates",
+              "data-live-search"            => "true",
+              "selectpicker-for-select-tag" => "",
+              :required                     => true,
+              "miq-select"                  => true,}
+        %option{"value" => "", "disabled" => ""}
+          = "<#{_('Choose')}>"
+      %span.help-block{"ng-show" => "angularForm.playbook_catalog_item.$error.required"}
+        = _("Required")
+  .form-group{"ng-if" => "vm.customButtonModel.button_type == 'ansible_playbook'"}
+    %label.control-label.col-md-2
+      = _("Inventory")
+    .col-md-8
+      %input{:type => "radio", :id => "inventory_localhost", :value => "localhost", 'ng-model' => 'vm.inventory'}
```

Comment:
We now have those 3 options in:

```
app/views/static/generic_object/main_custom_button_form.html.haml
44:      = label_tag('inventory_localhost', _("Localhost"), "title" => _('Run on localhost'))
47:      = label_tag('inventory_event_target', _("Target Machine"), "title" => _('Run on the target of the Policy Event'))
50:      = label_tag(_("Specific Hosts"))

app/views/shared/_playbook_options.html.haml
24:    = label_tag('inventory_localhost', _("Localhost"), "title" => _('Run on localhost'))
29:    = label_tag('inventory_event_target', _("Target Machine"), "title" => _('Run on the target of the Policy Event'))
34:    = label_tag('inventory_manual', _("Specific Hosts"), "title" => _('Enter a comma separated list of IP or DNS names'))

app/views/miq_policy/_action_options.html.haml
341:          = label_tag('inventory_localhost', _("Localhost"), "title" => _('Run on localhost'))
347:          = label_tag('inventory_event_target', _("Target Machine"), "title" => _('Run on the target of the Policy Event'))
353:          = label_tag('inventory_manual', _("Specific Hosts"), "title" => _('Enter a comma separated list of IP or DNS names'))
```

Time for a partial, I think ;)

Reply:
Also, the inputs are missing `name="inventory"`. Otherwise, you have 3 unrelated radio buttons, instead of one radio button group.